st2common/st2common/runners/python_action_wrapper.py

```diff
@@ -261,12 +264,26 @@ def _get_action_instance(self):
     args = parser.parse_args()
 
     config = json.loads(args.config) if args.config else {}
-    parameters = args.parameters
-    parameters = json.loads(parameters) if parameters else {}
     user = args.user
     parent_args = json.loads(args.parent_args) if args.parent_args else []
     log_level = args.log_level
 
+    parameters = {}
+
+    if args.parameters:
+        LOG.debug('Getting parameters from argument')
+        args_parameters = args.parameters
+        args_parameters = json.loads(args_parameters) if args_parameters else {}
+        parameters.update(args_parameters)
+
+    if args.stdin_parameters:
+        LOG.debug('Getting parameters from stdin')
+        stdin_parameters = json.loads(input())
```

Comment:
I wonder if there is some kind of timeout we could use here in case input is not provided in some reasonable time frame.

Shouldn't be an issue for action runner actions, but when you are debugging and running things manually.